hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,266 @@
+
+        Args:
+            command_string (str) : shell commands required to make changes to
+                                   openvswitch
+        Raises: SwitchError
+        Returns: If successful returns None else logs error message
+        """
+        args = shlex.split(command_string)
+        try:
+            subprocess.check_call(args)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError
+
+    def revert_port(self, port):
+        """Resets the port to the factory default.
+        Args:
+            port: Valid switch port
+
+        Returns: if successful returns None else error message.
+        """
+        import pdb; pdb.set_trace()
+
+        shell_cmd_1 = 'sudo ovs-vsctl del-port {port}'.format(port=port)
+        shell_cmd_2 = 'sudo ovs-vsctl add-port {switch} {port}'.format(
+                    switch=self.hostname, port=port
+                    ) + ' vlan_mode=native-untagged'
+        args_1 = shlex.split(shell_cmd_1)
+        args_2 = shlex.split(shell_cmd_2)
+        try:
+            subprocess.check_call(args_1)
+            subprocess.check_call(args_2)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError
+        
+#        retcode_1 = self.ovs_connect(shell_cmd_1)
+#        retcode_2 = self.ovs_connect(shell_cmd_2)
+#        if(retcode_1 is None and retcode_2 is None):
+#            return None
+
+    def modify_port(self, port, channel, new_network):
+        """ Changes vlan assignment to the port.
+
+        Args:
+            port: switch port in a valid format
+            channel: eg 'vlan/native' or 'vlan/<vlan_id>
+            new_network: vlan_id
+        Returns: If successful returns None else error.
+        """
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                return self._remove_native_vlan(interface)
+            else:
+                return self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                return self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                return self._add_vlan_to_trunk(interface, vlan_id)
+
+# 2. Private Methods:
+
+    def _string_to_list(self, a_string):
+        """Converts a string representation of list to list.
+        Args:
+            a_string: list output recieved as string.
+        Returns: object of list type.
+                 Empty list is put as None.
+        """
+        if a_string[0] == '[' and a_string[1] == ']':
```

Comment:
could simplify this to
```python
if a_string[] == '[]':
```

Reply:
suggestion accepted. next commit fixes it.